### PR TITLE
Update from my fork

### PR DESCRIPTION
**Pull request by supertobi**
New script to open an image geo location in gnome-maps: geo_uri.lua
Update of slideshowMusic.lua for darktable 2.0

### CODE REVIEW

File: contrib/geo_uri.lua [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+WARNING
+This script is not testet.
+
+USAGE
+* require this script from your main lua file
+* register a shortcut
+]]
+
+local dt = require "darktable"
+dt.configuration.check_version(...,{2,0,1})
```
Comment (boucman) [inline contrib/geo_uri.lua]: this won't work with api 3.0.0 (dt 2.0)

File: contrib/geo_uri.lua [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    return
+  end	
+    
+  local sel_images = dt.gui.selection()
+  
+  local lat1 = 0;
+  local lon1 = 0;
+  local i = 0;
+  
+  local sel_images = dt.gui.selection()
```
Comment (boucman) [inline contrib/geo_uri.lua]: this variable is defined twice (see line 54)

File: contrib/geo_uri.lua [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  
+  -- Use the first image with geo information
+  for _,image in ipairs(sel_images) do
+    if ((image.longitude and image.latitude) and 
+        (image.longitude ~= 0 and image.latitude ~= 90) -- Sometimes the north-pole but most likely just wrong data
+       ) then
+        
+        i = i + 1;
+        
+        if (i == 1) then
```
Comment (boucman) [inline contrib/geo_uri.lua]: please use the "break" statement instead of the variable "i" it will be cleaner

File: contrib/geo_uri.lua [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          lat1 = image.latitude;
+          lon1 = image.longitude;
+        end
+  
+      end
+    end
+
+    if (lat1 and lon1) then
```
Comment (boucman) [inline contrib/geo_uri.lua]: these variables are initialized to 0 (and not nil)

in lua, 0 is true, (only nil and false are false. 0 and "" are true) so this expression is always true

File: contrib/geo_uri.lua [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+--[[
+darktable script to open a geo uri in gnome-maps
+
+ADDITIANAL SOFTWARE NEEDED FOR THIS SCRIPT
+* gnome-maps >= 3.20
+
+WARNING
+This script is not testet.
```
Comment (boucman) [inline contrib/geo_uri.lua]: then please test it before I merge it :)

File: contrib/geo_uri.lua [hunk 6]
```diff
@@ -0,0 +1,92 @@
+--[[
+    This file is part of darktable,
+    Copyright 2014 by Tobias Jakobs.
```
Comment (boucman) [inline contrib/geo_uri.lua]: 2014 ?

**Comment by boucman** [comment]
interesting script overall, and a nice one to have... i'd rather have the update to slideshow.lua be a different pull request, but it doesn't really matter, i'll cherry pick those commits to the stable branch separately...

**Comment by supertobi** [comment]
Lets close this. I'll clean this up.

**Pull request closed without merging**